library_generation/utils/utilities.sh

```diff
@@ -179,30 +165,19 @@ download_tools() {
     export grpc_path=$(download_grpc_plugin "${grpc_version}" "${os_architecture}")
   fi
 
-  popd
-}
-
-download_generator_artifact() {
-  local gapic_generator_version=$1
-  local artifact=$2
-  local project=${3:-"gapic-generator-java"}
-  if [ ! -f "${artifact}" ]; then
-    # first, try to fetch the generator locally
-    local local_fetch_successful
-    local_fetch_successful=$(copy_from "$HOME/.m2/repository/com/google/api/${project}/${gapic_generator_version}/${artifact}" \
-      "${artifact}")
-    if [[ "${local_fetch_successful}" == "false" ]];then 
-      # download gapic-generator-java artifact from Google maven central mirror if not
-      # found locally
-      >&2 echo "${artifact} not found locally. Attempting a download from Maven Central"
-      download_from \
-      "https://maven-central.storage-download.googleapis.com/maven2/com/google/api/${project}/${gapic_generator_version}/${artifact}" \
-      "${artifact}"
-      >&2 echo "${artifact} found and downloaded from Maven Central"
-    else
-      >&2 echo "${artifact} found copied from local repository (~/.m2)"
-    fi
+  # prepare gapic-generator-java
+  # We will assume the generator location environment variable will point to the
+  # location of the generator JAR and then we will simply copy it. Although
+  # these env vars are already validated upstream in the python scripts, we
+  # still validate them here for testing purposes
+  if [[ -z "${DOCKER_GAPIC_GENERATOR_LOCATION}" ]]; then
+    >&2 echo "Env var DOCKER_GAPIC_GENERATOR_LOCATION must be set and pointing to the generator jar"
+    exit 1
+  else
+    >&2 echo "Using gapic-generator-java version baked into the container"
+    cp "${DOCKER_GAPIC_GENERATOR_LOCATION}" "${output_folder}"
```

Comment:
Can we just use the jar as it is without copying? Is it for `gapic-generator-java-wrapper`?

Reply:
The wrapper assumes the jar to be in CWD

https://github.com/googleapis/sdk-platform-java/blob/58da069a99a7240cbec18b7dbdb010fb74c4d551/library_generation/gapic-generator-java-wrapper#L6

I'm not sure if we can both avoid copying the jar and at the same time not have a `DOCKER_GAPIC_GENERATOR_LOCATION` variable. We need a way to know where the jar is and make it compatible with local development as well.

Maybe we can copy the jar directly into the home `.m2` folder? We would still be copying it just like we've been doing so far.